api/app/src/external/fhir/shared/index.ts

```diff
@@ -49,7 +49,26 @@ export function isoDateRangeToFHIRDateQuery(dateFrom?: string, dateTo?: string):
   return fhirDateQuery;
 }
 
+const resourcesSupportingDateQueries: { [index: string]: boolean } = {
```

Comment:
This was intentional since in theory map has a quicker lookup than arrays

Reply:
💯 , maps are much faster for element access based on key. But since we have so few elements, one might say is negligible.

One issue with the `map<string, boolean>` approach is there's no compilation check on the items, so if someone mistype a letter, we won't be notified at compile time.

If we change the type declaration to `{ [k in MedplumResourceType]?: boolean }`, we're gold.

Could even make the value always `true`, but that's defs NABD.

Reply:
note that `{ [k in MedplumResourceType]?: boolean }` doesn't work as it expects all of the literals to be present in the map - will look into this

Reply:
I works on my computer:

<img width="636" alt="image" src="https://github.com/metriport/metriport/assets/2132564/8d970f3d-d6f8-4f14-97d2-ae514e054cef">

When I remove the question mark then it doesn't, as expected:

<img width="639" alt="image" src="https://github.com/metriport/metriport/assets/2132564/abd037af-43e1-4534-ba3f-866e73429219">

This is not a big deal, I'm reporting this mostly so we check whether our envs are matching well. I did `npm install` and `npm run build` to make sure it was not just VSCode being lenient.

Reply:
Ah I missed the `?` in the original comment - fixed